app/controllers/observations/inat_imports_controller.rb

```diff
@@ -0,0 +1,190 @@
+# frozen_string_literal: true
+
+# import iNaturalist Observations as MO Observations
+# Actions
+# -------
+# new (get)
+# create (post)
+# authorization_response (get)
+#
+# Work flow:
+# 1. User calls `new`, fills out form
+# 2. create
+#      saves some user data in a InatImport instance
+#        attributes include: user, inat_ids, token, state
+#    passes things off (redirects) to iNat at the inat_authorization_url
+# 3. iNat
+#    checks if MO is authorized to access iNat user's confidential data
+#      if not, asks iNat user for authorization
+#    iNat calls the MO redirect_url (authorization_response) with a code param
+# 4. MO continues in the authorization_response action
+#    Reads the saved InatImport instance
+#    Updates the InatImport instance with the code received from iNat
+#    Enqueues an InatImportJob, passing in the InatImport instance
+# 5. The rest happens in the background. The InatImportJob:
+#      Uses the `code` to obtain an oauth access_token
+#      Trades the oauth token for a JWT api_token
+#      Makes an authenticated iNat API request for the desired observations
+#      For each iNat obs in the results,
+#         creates an Inat::Obs
+#         adds an MO Observation, mapping Inat::Obs details to the MO Observation
+#         adds Inat photos to the MO Observation via the MO API
+#
+module Observations
+  class InatImportsController < ApplicationController
+    before_action :login_required
+    before_action :pass_query_params
+
+    # Site for authorization and authentication requests
+    SITE = "https://www.inaturalist.org"
+    # what iNat will call after user responds to authorization request
+    REDIRECT_URI =
+      "http://localhost:3000/observations/inat_imports/authorization_response"
+    # iNat's id for the MO application
+    APP_ID = Rails.application.credentials.inat.id
+    # The iNat API. Not called here, but reference in tests and ActiveJob
+    API_BASE = "https://api.inaturalist.org/v1"
+
+    def new; end
+
+    def create
```

Comment:
For this ABC metrics violation, I would start with moving all the error checking into it's own method so it could be a single lline in this method.